spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -302,74 +293,39 @@ public Transform unknown(int fieldId, String sourceName, int sourceId, String tr
     return transforms.stream().filter(Objects::nonNull).toArray(Transform[]::new);
   }
 
-  public static Distribution buildRequiredDistribution(org.apache.iceberg.Table table) {
```

Comment:
I moved the distribution and order specific logic into a separate class as we will also handle copy-on-write and merge-on-read scenarios.